src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,328 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2024 SECO Mind Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Button, Card, Col, Container, Form, Row, Spinner, Table } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import {
+  AstarteDataTuple,
+  AstarteDataTreeNode,
+  AstartePropertyData,
+  AstarteDatastreamIndividualData,
+  AstarteDatastreamObjectData,
+} from 'astarte-client';
+import _ from 'lodash';
+
+import BackButton from './ui/BackButton';
+import Empty from './components/Empty';
+import WaitForData from './components/WaitForData';
+import { useAstarte } from './AstarteManager';
+import { AlertsBanner, useAlerts } from 'AlertManager';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const MAX_SHOWN_VALUES = 20;
+
+function formatAstarteData(data?: AstarteDataTuple): string {
+  const value = data?.value;
+  if (value == null) {
+    return '';
+  }
+  if (_.isArray(value)) {
+    return JSON.stringify(value);
+  }
+  if (_.isBoolean(value)) {
+    return value ? 'true' : 'false';
+  }
+  if (_.isNumber(value)) {
+    return value.toString();
+  }
+  if (_.isNull(value)) {
+    return '';
+  }
+  return String(value);
+}
+
+interface PropertyTreeProps {
+  treeNode: AstarteDataTreeNode<AstartePropertyData>;
+}
+
+const PropertyTree = ({ treeNode }: PropertyTreeProps): React.ReactElement => (
+  <pre>
+    <code>{JSON.stringify(treeNode.toLastValue(), null, 2)}</code>
+  </pre>
+);
+
+interface IndividualDatastreamTableProps {
+  treeNode: AstarteDataTreeNode<AstarteDatastreamIndividualData>;
+}
+
+const IndividualDatastreamTable = ({
+  treeNode,
+}: IndividualDatastreamTableProps): React.ReactElement => {
+  const dataValues = treeNode.toLinearizedData();
+  if (dataValues.length === 0) {
+    return <p>No data sent by the device.</p>;
```

Comment:
Let's adapt this sentence because it may be misunderstood.
For example, maybe the divice published data before but not in the selected timeframe